agent/http_register.go

```diff
@@ -94,6 +94,7 @@ func init() {
 	registerEndpoint("/v1/internal/ui/nodes", []string{"GET"}, (*HTTPServer).UINodes)
 	registerEndpoint("/v1/internal/ui/node/", []string{"GET"}, (*HTTPServer).UINodeInfo)
 	registerEndpoint("/v1/internal/ui/services", []string{"GET"}, (*HTTPServer).UIServices)
+	registerEndpoint("/v1/internal/ui/terminating-gateway-services/", []string{"GET"}, (*HTTPServer).TerminatingGatewayServices)
```

Comment:
Random thought: Can this API be merged with the `/v1/internal/ui/services` endpoint above? That API does a similar thing in that it returns information about what services another service is a proxy for with the `ProxyFor` field.

Adding gateways to this endpoint might be some work since currently the endpoint can just use service definition information to summarize data, but with gateways we would need to look at config entries as well.

Reply:
That endpoint supports filtering and we could also plumb through the ability to specify a service kind (the RPC it makes already allows for this). That way we could reuse the same endpoint.

Reply:
The PR has been de-scoped to not add HTTP endpoints atm, but agreed that it makes sense to reuse the endpoint. 👍